core/tx_test.go

```diff
@@ -0,0 +1,40 @@
+package core
+
+import (
+	"io/ioutil"
+	"log"
+	"os"
+	"testing"
+
+	"github.com/BOPR/common"
+	"github.com/jinzhu/gorm"
+	"github.com/stretchr/testify/assert"
+	"gorm.io/driver/sqlite"
+)
+
+func TestPopTx(t *testing.T) {
+	tmpfile, err := ioutil.TempFile("", "test.db")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	defer os.Remove(tmpfile.Name())
+
+	db, err := gorm.Open("sqlite", sqlite.Open("gorm.db"))
```

Comment:
There is a error thrown here that wasn't caught due to which db is nil and hence the nil pointer dereference error. 

```
invalid database source: &{gorm.db} is not a valid type
```

Above is what I get when I log it

Reply:
Thanks. It turns out I'm using sqlite from gorm V2 but our codebase uses gorm V1